vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/credentialverifier/validator/MsoMdocValidator.kt

```
@@ -5,73 +5,174 @@ import co.nstant.`in`.cbor.model.MajorType
-                throw ValidationException(ERROR_MESSAGE_INVALID_VALID_UNTIL_MSO, ERROR_CODE_INVALID_VALID_UNTIL_MSO)
-            }
+            val mso = credentialData.mso
 
-            if(!isValidUntilGreaterThanValidFrom){
-                logger.severe("Error while doing validity verification - invalid validFrom / validUntil in the MSO of the credential")
-                throw ValidationException(ERROR_MESSAGE_INVALID_DATE_MSO, ERROR_CODE_INVALID_DATE_MSO)
-            }
+            validateMsoStructure(mso)
+            validateValidityInfo(mso, isLatest)
             return true
         } catch (exception: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)